makes/entries.sh

```diff
@@ -7,13 +7,46 @@ set -e -o pipefail
 base=$(realpath "$(dirname "$0")/..")
 
 mkdir -p "${base}/target/entries-logs"
-while IFS= read -r sh; do
+
+declare -a pids=()
+declare -a test_names=()
+declare -a failed_tests=()
+
+run_test() {
+    local sh="$1"
+    local log_name="${sh//\//__}"
     mkdir -p "${base}/target/${sh}"
-    if /bin/bash -c "cd \"target/${sh}\" && exec \"${base}/entries/${sh}\" \"${base}\" > \"${base}/target/entries-logs/${sh}.txt\" 2>&1"; then
+    if ( cd "${base}/target/${sh}" && "${base}/entries/${sh}" "${base}" ) > "${base}/target/entries-logs/${log_name}.txt" 2>&1; then
         echo "👍🏻 ${sh} passed"
+        return 0
     else
-        cat "${base}/target/entries-logs/${sh}.txt"
         echo "❌ ${sh} failed"
-        exit 1
+        return 1
     fi
-done < <( find "${base}/entries" -name '*.sh' -exec basename {} \; )
+}
+while IFS= read -r -d '' sh; do
+    sh="${sh#./}"
+    run_test "$sh" &
+    pids+=($!)
+    test_names+=("$sh")
+done < <( cd "${base}/entries" && find . -type f -name '*.sh' -print0 )
+
+# Wait for all tests to complete and collect results
+exit_code=0
+for i in "${!pids[@]}"; do
+    if ! wait "${pids[$i]}"; then
+        failed_tests+=("${test_names[$i]}")
+        exit_code=1
+    fi
+done
```

Comment:
@vbvictor I believe, you can do the same, but with the help of [GNU Parallel](https://www.gnu.org/software/parallel/). The code will be much shorter.

Reply:
@vbvictor yes, it's fine. Just don't forget to document it in the README.